File: backend/app/Models/BotSource.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class BotSource extends Model
{
    protected $fillable = ['bot_id', 'type', 'title', 'content', 'url', 'file_name', 'file_path', 'file_type', 'file_size', 'status', 'error_message', 'meta',];
    protected $casts = ['meta' => 'array',];
    public function bot()
    {
        return $this->belongsTo(Bot::class);
    }
   
}

File: fastapi/app/models/bot_source.py
from pydantic import BaseModel, ConfigDict


class BotSource(BaseModel):
    """
    Represents a knowledge source attached to a bot.
    """

    model_config = ConfigDict(
        extra="forbid",
        frozen=True,
    )

    id: int

    type: str

    title: str

    content: str | None = None

    url: str | None = None

    file_name: str | None = None

    file_path: str | None = None

    file_url: str | None = None

    file_type: str | None = None

    file_size: int | None = None

    status: str